src/Scaffolding/VS.Web.CG.Mvc/Templates/MinimalApi/MinimalApiNoClass.cshtml

```diff
@@ -1,103 +1,105 @@
-        @:.WithName("@deleteModel")
-        @:.Produces<@modelName>(StatusCodes.Status200OK);
-    }
-    else
-    {
-        @:.WithName("@deleteModel");  
-    }
-    }
-    }
+@inherits Microsoft.VisualStudio.Web.CodeGeneration.Templating.RazorTemplateBase
+@{
+    string modelName = Model.ModelType.Name;
+    string modelConstructor = $"{modelName}()";
+    string modelArray = $"{modelName}[]";
+    string routePrefix = "/api/" + modelName;
+    string endPointsClassName = Model.EndpointsName;
+    string methodName = $"Map{@modelName}Endpoints";
+    string pluralModel = $"{@modelName}s";
+    string getAllModels = $"GetAll{@pluralModel}";
+    string getModelById = $"Get{@modelName}ById";
+    string deleteModel = $"Delete{@modelName}";
+    string createModel = $"Create{@modelName}";
+    string updateModel = $"Update{@modelName}";
+    string resultsExtension = (Model.OpenAPI ? "TypedResults" : "Results") + ".NoContent()";
+}
+ 
+   public static void @Model.MethodName (this IEndpointRouteBuilder routes)
+    {
+        var group = routes.MapGroup("@routePrefix");
+        group.MapGet("/", () =>
+        {
+            return new [] { new @modelConstructor };
+        })
+    @{
+    if(Model.OpenAPI)
+    {
+        @:.WithTags(nameof(@modelName))
+        @:.WithName("@getAllModels")
+        @:.Produces<@modelArray>(StatusCodes.Status200OK);
+    }
+    else
+    {
+        @:.WithName("@getAllModels");
+    }
+    }
+
+        group.MapGet("/{id}", (int id) =>
+        {
+            //return new @modelName { ID = id };
+        })
+    @{
+    if(Model.OpenAPI)
+    {
+        @:.WithTags(nameof(@modelName))
+        @:.WithName("@getModelById")
+        @:.Produces<@modelName>(StatusCodes.Status200OK);
+    }
+    else
+    {
+        @:.WithName("@getModelById");
+    }
+    }
+
+        group.MapPut("/{id}", (int id, @modelName input) =>
+        {
+            return @resultsExtension;
+        })
+    @{
+    if(Model.OpenAPI)
+    {
+        @:.WithTags(nameof(@modelName))
+        @:.WithName("@updateModel")
+        @:.Produces(StatusCodes.Status204NoContent);
+    }
+    else
+    {
+        @:.WithName("@updateModel");
+    }
+    }
+
+        group.MapPost("/", (@modelName model) =>
+        {
+            //return Results.Created($"/@pluralModel/{model.ID}", model);
+        })
+    @{
+    if(Model.OpenAPI)
+    {
+        @:.WithTags(nameof(@modelName))
+        @:.WithName("@createModel")
+        @:.Produces<@modelName>(StatusCodes.Status201Created);
+    }
+    else
+    {
+        @:.WithName("@createModel");
+    }
+    }
+
+        group.MapDelete("/{id}", (int id) =>
+        {
+            //return Results.Ok(new @modelName { ID = id });
+        })
+    @{
+    if(Model.OpenAPI)
+    {
+        @:.WithTags(nameof(@modelName))
+        @:.WithName("@deleteModel")
+        @:.Produces<@modelName>(StatusCodes.Status200OK);
+    }
+    else
+    {
+        @:.WithName("@deleteModel");  
+    }

```

Comment:
This indentation seems off but I'm not enough of an expert on these templates to know!

Reply:
It aligns correctly with the group.Map closing parenthesis. Razor templating looks odd sometimes but seems good in my tests too.